controllers/discovery/searchDomain.js

```diff
@@ -20,32 +20,33 @@ const SearchDomain = async (req, res) => {
   try {
     const domains = await sequelize.query(
       `SELECT 
-                "Domain"."domain_page_id",
-                "Domain"."domain_name",
+        "Domain"."domain_page_id" AS "domain_id_followed",
+        "Domain"."domain_name",
+        "Domain"."logo",
+        "Domain"."short_description",
+        "Domain"."credder_score",
+        count("domainFollower"."user_id_follower") AS "followersCount",
+        (
+          SELECT "f"."user_id_follower" 
+          FROM "user_follow_domain" AS "f" 
+          WHERE "f"."user_id_follower" = :userId 
+          AND "f"."domain_id_followed" = "Domain"."domain_page_id"
+          LIMIT 1
+        ) AS "user_id_follower"
+      FROM "domain_page" AS "Domain" 
+      LEFT JOIN "user_follow_domain" AS "domainFollower" ON "Domain"."domain_page_id" = "domainFollower"."domain_id_followed" 
+      WHERE "Domain"."domain_name" ILIKE :likeQuery 
+      GROUP BY 
+        "user_id_follower",
+        "Domain"."domain_page_id",
+        "Domain"."domain_name",
 				"Domain"."logo",
 				"Domain"."short_description",
-                "Domain"."credder_score",
-                count("domainFollower"."user_id_follower") 
-                    AS "followersCount",
-                (SELECT "f"."user_id_follower" AS "user_id_follower" FROM "user_follow_domain" AS "f" WHERE "f"."user_id_follower" = :userId AND "f"."domain_id_followed" = "Domain"."domain_page_id")
-            FROM "domain_page" 
-                AS "Domain" 
-            LEFT JOIN "user_follow_domain" 
-                AS "domainFollower" 
-            ON "Domain"."domain_page_id" = 
-                "domainFollower"."domain_id_followed" 
-            WHERE 
-                "Domain"."domain_name" ILIKE :likeQuery 
-            GROUP BY 
-                "Domain"."domain_page_id",
-                "Domain"."domain_name",
-				"Domain"."logo",
-				"Domain"."short_description",
-                "Domain"."credder_score"
-            ORDER BY
-                "user_id_follower" ASC,
-                "followersCount" DESC
-            LIMIT :limit`,
+        "Domain"."credder_score"
+      ORDER BY
+        "user_id_follower" ASC,
+        "followersCount" DESC
+      LIMIT :limit`,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the SQL query are mostly about renaming and reordering columns, which should not affect the logic of the code. However, it's important to ensure that these changes align with the rest of your application, especially where this data is consumed. If other parts of your application rely on the old column names or order, those will need to be updated as well. 

One thing to note is that the `user_id_follower` is now being included in the GROUP BY clause. This could potentially change the results of your query if there are multiple rows with the same `domain_page_id` but different `user_id_follower`. Make sure this change aligns with your intended behavior.

Also, the `user_id_follower` is now being ordered in ascending order before `followersCount` in descending order. Ensure this change aligns with your intended sorting behavior.

Lastly, always remember to sanitize inputs to SQL queries to prevent SQL injection attacks. It seems like you're using parameterized queries (`:userId`, `:likeQuery`, `:limit`), which is good practice. Just make sure the values for these parameters are properly validated and sanitized before they are used in the query.


<!-- This is an auto-generated comment by OSS CodeRabbit -->